src/Controller/CreatePayPalOrderAction.php

```diff
@@ -18,36 +18,47 @@
 
 final class CreatePayPalOrderAction
 {
-    private Payum $payum;
-
-    private OrderRepositoryInterface $orderRepository;
-
-    private FactoryInterface $stateMachineFactory;
-
-    private ObjectManager $paymentManager;
-
-    private PaymentStateManagerInterface $paymentStateManager;
-
-    private OrderProviderInterface $orderProvider;
-
-    private CapturePaymentResolverInterface $capturePaymentResolver;
-
     public function __construct(
-        Payum $payum,
-        OrderRepositoryInterface $orderRepository,
-        FactoryInterface $stateMachineFactory,
-        ObjectManager $paymentManager,
-        PaymentStateManagerInterface $paymentStateManager,
-        OrderProviderInterface $orderProvider,
-        CapturePaymentResolverInterface $capturePaymentResolver
+        private readonly Payum $payum,
+        private readonly OrderRepositoryInterface $orderRepository,
+        private readonly FactoryInterface $stateMachineFactory,
+        private readonly ObjectManager $paymentManager,
+        private readonly PaymentStateManagerInterface $paymentStateManager,
+        private readonly OrderProviderInterface $orderProvider,
+        private readonly CapturePaymentResolverInterface $capturePaymentResolver,
     ) {
-        $this->payum = $payum;
-        $this->orderRepository = $orderRepository;
-        $this->stateMachineFactory = $stateMachineFactory;
-        $this->paymentManager = $paymentManager;
-        $this->paymentStateManager = $paymentStateManager;
-        $this->orderProvider = $orderProvider;
-        $this->capturePaymentResolver = $capturePaymentResolver;
+        trigger_deprecation(
```

Comment:
IMO it should be placed inside an `if` statement because there is no way to disable these deprecation notifications.

Reply:
Added `if` statements